cmd/spiffe-aws-assume-role/cli/cli.go

```diff
@@ -40,24 +53,41 @@ func (c *CredentialsCmd) Run() error {
 	return err
 }
 
-var CLI struct {
+type CLI struct {
 	Credentials CredentialsCmd `kong:"cmd,help:'print credentials in a format usable as an AWS credentials_process'"`
 }
 
-func Run(args []string) error {
-	p, err := kong.New(&CLI)
-	if err != nil {
-		return err
+func RunWithDefaultContext(args []string) error {
+	context := &CliContext{
+		JWTSourceProvider: credentials.StandardJWTSourceProvider,
+		STSProvider:       credentials.StandardSTSProvider,
 	}
 
-	ctx, err := p.Parse(args)
+	return Run(context, args)
+}
+
+func Run(context *CliContext, args []string) error {
```

Comment:
Instead of defaulting in the face of a nil/empty context I thought it would be more clear to add a helper function. This way the name is a dead giveaway that 1. there is a context and 2. it's being set to some default value.